File: ki-aim-platform/ki-aim-frontend/src/app/shared/model/data-type.ts
export enum DataType {
    BOOLEAN = "BOOLEAN",
	DATE_TIME = "DATE_TIME",
	DECIMAL = "DECIMAL",
	INTEGER = "INTEGER",
	STRING = "STRING",
	DATE = "DATE",
	UNDEFINED = "UNDEFINED",
}

export function dataTypeFromString(value: String): DataType {
	switch (value) {
		case "BOOLEAN": return DataType.BOOLEAN
		case "DATE_TIME": return DataType.DATE_TIME
		case "DECIMAL": return DataType.DECIMAL
		case "INTEGER": return DataType.INTEGER
		case "STRING": return DataType.STRING
		case "DATE": return DataType.DATE
		default: return DataType.UNDEFINED
	}
}

File: cinnamon-model/src/main/java/de/kiaim/cinnamon/model/enumeration/DataType.java
package de.kiaim.cinnamon.model.enumeration;

import lombok.Getter;

import java.util.Set;

/**
 * Data types supported by Cinnamon.
 *
 * @author Daniel Preciado-Marquez
 */
@Getter
public enum DataType {
	BOOLEAN(DataScale.NOMINAL, Set.of(DataScale.NOMINAL)),
	DATE_TIME(DataScale.DATE, Set.of(DataScale.DATE)),
	DECIMAL(DataScale.RATIO, Set.of(DataScale.NOMINAL, DataScale.ORDINAL, DataScale.INTERVAL, DataScale.RATIO)),
	INTEGER(DataScale.INTERVAL, Set.of(DataScale.NOMINAL, DataScale.ORDINAL, DataScale.INTERVAL)),
	STRING(DataScale.NOMINAL, Set.of(DataScale.NOMINAL, DataScale.ORDINAL)),
	DATE(DataScale.DATE, Set.of(DataScale.DATE)),
	UNDEFINED(DataScale.NOMINAL, Set.of());

	DataType(final DataScale defaultScale, final Set<DataScale> supportedScales) {
		this.defaultScale = defaultScale;
		this.supportedScales = supportedScales;
	}

	/**
	 * The default data sale for the data type.
	 */
	private final DataScale defaultScale;

	/**
	 * Set of data scales which can be used in combination with the data type.
	 */
	private final Set<DataScale> supportedScales;
}
